initializer_test.go

```diff
@@ -0,0 +1,84 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ucfg
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type structInitializer struct {
+	I int
+	J int
+}
```

Comment:
we should add tests for non-struct types as well. For example primitive types or maps:

```
type negInt int

func (i *negInt) InitDefaults() {
  *i = -1
}
```

Although I don't think maps are a common use-case, custom primitive types to act as enums are rather common.